tools/crm_error.c

```diff
@@ -57,11 +56,99 @@ static GOptionEntry entries[] = {
     { NULL }
 };
 
+PCMK__OUTPUT_ARGS("result-code", "int", "char *", "char *");
+static int
+result_code_default(pcmk__output_t *out, va_list args)
+{
+    int code = va_arg(args, int);
+    char *name = va_arg(args, char *);
+    char *desc = va_arg(args, char *);
+
+    long long most_negative = 0;
+    int code_width = 0;
+
+    if (out->is_quiet(out)) {
+        /* If out->is_quiet(), don't print the code. Print name and/or desc in a
+         * compact format for text output, or print nothing at all for none-type
+         * output.
+         */
+        if (pcmk__str_eq(out->fmt_name, "text", pcmk__str_none)) {
+            if ((name != NULL) && (desc != NULL)) {
+                pcmk__formatted_printf(out, "%s - %s\n", name, desc);
+
+            } else if ((name != NULL) || (desc != NULL)) {
+                pcmk__formatted_printf(out, "%s\n",
+                                       ((name != NULL)? name : desc));
+            }
+        }
+        return pcmk_rc_ok;
+    }
+
+    /* Get length of longest (most negative) standard Pacemaker return code
+     * This should be longer than all the values of any other type of return
+     * code.
+     */
+    most_negative = pcmk_rc_error - (long long) pcmk__n_rc + 1;
+    code_width = (int) snprintf(NULL, 0, "%lld", most_negative);
+
+    if ((name != NULL) && (desc != NULL)) {
+        int name_width = 0;
+
+        // Get length of longest standard Pacemaker return code name
+        for (int lpc = 0; lpc < pcmk__n_rc; lpc++) {
+            int len = (int) strlen(pcmk_rc_name(pcmk_rc_error - lpc));
+            name_width = QB_MAX(name_width, len);
+        }
+        return out->info(out, "% *d: %-*s  %s", code_width, code, name_width,
+                         name, desc);
+    }
```

Comment:
We don't have to calculate code_width and name_width every time, we can make them static variables